Add standalone Daily + Twilio SIP example

markbackman opened the pull request:
#### Please describe the changes in your PR. If it is addressing an issue, please reference that as well.

This removes the Daily SIP + Twilio example from the phone-chatbot directory. The goal in doing so is to provide a clear, simple example of how to set up a Daily SIP + Twilio dial-in call.

vr000m commented on examples/phone-chatbot-daily-twilio-sip/bot.py:
```diff
@@ -0,0 +1,157 @@
+    async def on_dialin_ready(transport, cdata):
+        logger.info(f"Forwarding call {call_id} to {sip_uri}")
+
+        try:
```
to make it it future proof, we should only fire execute this for one sip endpoint, if multiple are configured.

markbackman replied:
Good idea! This really hung me up, so we should make it clear.

vr000m replied:
also the sip_uri is the cdata

vr000m commented on examples/phone-chatbot-daily-twilio-sip/server.py:
```diff
@@ -0,0 +1,111 @@
+        # Generate TwiML response to put the caller on hold with music
+        resp = VoiceResponse()
+        resp.play(
+            url="http://com.twilio.sounds.music.s3.amazonaws.com/MARKOVICHAMP-Borghestral.mp3",
```
To avoid transitions from ringing to hold music to ringing, we can use the  US ringtone. The following can be converted to an mp3 file https://commons.wikimedia.org/wiki/File:US_ringback_tone.ogg

markbackman replied:
Updated! This is nice. I had to use the Twilio media storage to make this work seemlessly.

vr000m reviewed (approved):
LGTM some comments inline

vr000m commented on examples/phone-chatbot-daily-twilio-sip/bot.py:
```diff
@@ -0,0 +1,157 @@
+        except Exception as e:
+            logger.error(f"Failed to forward call: {str(e)}")
+            raise
+
```
```suggestion
    @transport.event_handler("on_dialin_connected")
    async def on_dialin_connected(transport, data):
        logger.debug(f"Dial-in connected: {data}")

    @transport.event_handler("on_dialin_stopped")
    async def on_dialin_stopped(transport, data):
        logger.debug(f"Dial-in stopped: {data}")

    @transport.event_handler("on_dialin_error")
    async def on_dialin_error(transport, data):
        logger.error(f"Dial-in error: {data}")
        # If there is an error, the bot should leave the call 
        # Maybe handled in on_participant_left
        # await task.cancel()

    @transport.event_handler("on_dialin_warning")
    async def on_dialin_warning(transport, data):
        logger.warning(f"Dial-in warning: {data}")
```

markbackman replied:
@vr000m what was your comment on this one?

vr000m replied:
These error handlers may help diagnose issues and for example safely close the bot when an error happens. It is very likely user-left will fire anyway, but good to know thatthe call failed because the telephony dropped.

connected maybe useful if someone passes custom sip-headers, for example, to/from addresses from twilio were passed in the sip address. e.g., room-name@sip-hostname.com?x-custom-header=custom-val

markbackman commented on examples/phone-chatbot-daily-twilio-sip/server.py:
```diff
@@ -0,0 +1,111 @@
+
+        # Create a Daily room with SIP capabilities
+        try:
+            room_details = await create_sip_room(request.app.session)
```
👍

vr000m commented on examples/phone-chatbot-daily-twilio-sip/bot.py:
```diff
@@ -0,0 +1,183 @@
+twilio_client = Client(os.getenv("TWILIO_ACCOUNT_SID"), os.getenv("TWILIO_AUTH_TOKEN"))
+
+
+async def run_bot(room_url: str, token: str, call_id: str, sip_uri: str) -> None:
```
if there are multiple sip addresses, you may have to pass this as an array. Or maybe does not matter, because dialin-ready will fire for each configured sip address,

markbackman replied:
> Or maybe does not matter, because dialin-ready will fire for each configured sip address,

Correct. It will fire for each address.

markbackman commented:
@DominicStewart In my last commit, I removed the Twilio parts from the phone-chatbot directory. Can you confirm that I removed the necessary code? I'll wait for you to confirm before I merge this.
